### ISSUE TITLE
Installation wizard - Node setup test buttons - Cosmos base-chain #199

### ISSUE THREAD

**Issue by zimaah** [SimplyVC/panic#206]
### User Story

As a node operator, I want to be able to verify that the various URLs requested are correctly configured before my settings are finalised.

### Description

The scope of this task is **not** limited exclusively to the frontend aspect of test buttons associated with the nodes setup step - see #48.

### Requirements

Introduce a Test button for the following per node added:

- (1) Cosmos Rest Server
- When the user clicks on the Test button corresponding to the Cosmos Rest Server and the connection cannot be established show a message reading, "**Cannot connect with Cosmos Rest URL - _user URL input_. Please review your input and try again**"
- When the user clicks on the Test button and the connection is established show a message reading, "**Successfully connected**"
- (2) Prometheus Endpoint URL
- When the user clicks on the Test button corresponding to the Prometheus Endpoint URL and the connection cannot be established show a message reading, "**Cannot connect with Prometheus Endpoint URL - _user URL input_. Please review your input and try again**"
- When the user clicks on the Test button corresponding to the Prometheus Endpoint URL and the connection is established show a message reading, "**Successfully connected**"
- (3) Node Exporter URL 
- When the user clicks on the Test button corresponding to the Node Exporter URL and the connection cannot be established show a message reading, "**Cannot connect with Node Exporter URL - _user URL input_. Please review your input and try again**"
- When the user clicks on the Test button corresponding to the Node Exporter URL and the connection is established show a message reading, "**Successfully connected**"
- (4) Tendermint RPC URL
- When the user clicks on the Test button corresponding to the Tendermint RPC URL and the connection cannot be established show a message reading, "**Cannot connect with Tendermint RPC URL - _user URL input_. Please review your input and try again**"
- When the user clicks on the Test button corresponding to the Tendermint RPC URL and the connection is established show a message reading, "**Successfully connected**"
- Use toast messages to display test feedback. Timeout = 5s
- Handling connection timeout of 3s.

### Blocked by

Not dependent on another ticket to be tackled.

### Acceptance criteria

**Scenario**: Node operator is using the installation wizard to complete node setup for a Cosmos based chain
**Given**: The node operator is setting up the Cosmos Rest Server
**Then**: The node operator is presented with a Test button corresponding to Cosmos rest server
**And**: If the node operator clicks on the Test button and the input URL is invalid
**Then**: Display a toast message reading, "Cannot connect with Cosmos Rest URL - _user URL input_. Please review your input and try again"
**And**: The toast message disappears after 5s
**And**: If the node operator clicks on the Test button and the input URL is valid
**Then**:  Display a toast message reading, "Successfully connected"
**And**: The toast message disappears after 5s

**Scenario**: Node operator is using the installation wizard to complete node setup for a Cosmos based chain
**Given**: The node operator is setting up the Node Exporter 
**Then**: The node operator is presented with a Test button corresponding to Node Exporter URL 
**And**: If the node operator clicks on the Test button and the input URL is invalid
**Then**: Display a toast message reading, "Cannot connect with Node Exporter URL - _user URL input_. Please review your input and try again"
**And**: The toast message disappears after 5s
**And**: If the node operator clicks on the Test button and the input URL is valid
**Then**:  Display a toast message reading, "Successfully connected"
**And**: The toast message disappears after 5s

**Scenario**: Node operator is using the installation wizard to complete node setup for a Cosmos based chain
**Given**: The node operator is setting up the Prometheus Endpoint
**Then**: The node operator is presented with a Test button corresponding to Prometheus Endpoint URL
**And**: If the node operator clicks on the Test button and the input URL is invalid
**Then**: Display a toast message reading, "Cannot connect with Prometheus Endpoint URL - _user URL input_. Please review your input and try again"
**And**: The toast message disappears after 5s
**And**: If the node operator clicks on the Test button and the input URL is valid
**Then**:  Display a toast message reading, "Successfully connected"
**And**: The toast message disappears after 5s

**Scenario**: Node operator is using the installation wizard to complete node setup for a Cosmos based chain
**Given**: The node operator is setting up the Tendermint RPC
**Then**: The node operator is presented with a Test button corresponding to Tendermint RPC URL
**And**: If the node operator clicks on the Test button and the input URL is invalid
**Then**: Display a toast message reading, "Cannot connect with Tendermint RPC URL - _user URL input_. Please review your input and try again"
**And**: The toast message disappears after 5s
**And**: If the node operator clicks on the Test button and the input URL is valid
**Then**:  Display a toast message reading, "Successfully connected"
**And**: The toast message disappears after 5s

**Scenario**: Node operator is using the installation wizard to complete node setup for a Cosmos based chain
**Given**: The node operator is setting up any service, for example Cosmos REST
**Then**: The node operator is presented with a Test button corresponding to Cosmos REST URL
**And**: If the node operator clicks on the Test button and their connection does not allow them to access the node
**And**: No ping is recieved within the stipulated time (3 secons)
**Then**: Display a toast message reading, "Error: Connection timeout - Please make sure you are connected to the correct network"
**And**: The toast message disappears after 5s
**And**: If the node operator clicks on the Test button and the input URL is valid
**Then**:  Display a toast message reading, "Successfully connected"
**And**: The toast message disappears after 5s

**Comment by zimaah** [comment]
I'm unblocking this ticket because the solution has been verified and explained on Slack.

**Comment by itsciccio** [comment]
Using `fetch()` calls from the frontend raises an issue when pinging Prometheus, Node Exporter and Tendermint RPC. Only works with Cosmos REST.
